e2e/test/scenarios/onboarding/setup/user_settings.cy.spec.js

```diff
@@ -14,42 +15,59 @@ describe("user > settings", () => {
   });
 
   it("should be able to remove first name and last name (metabase#22754)", () => {
-    cy.visit("/account/profile");
-    cy.findByText(fullName);
-    cy.findByLabelText("First name").clear();
-    cy.findByLabelText("Last name").clear();
-    cy.button("Update").click();
+    const page = AccountSettingsPage.visit();
+
+    page.header.verifyFullname(fullName);
+    page.tabs.profileForm.fill({ firstName: "", lastName: "" }).submit();
 
     cy.reload();
 
-    cy.findByLabelText("First name").should("be.empty");
-    cy.findByLabelText("Last name").should("be.empty");
+    page.tabs.profileForm.verifyValues({ firstName: "", lastName: "" });
   });
 
   it("should show user details with disabled submit button", () => {
-    cy.visit("/account/profile");
-    cy.findByTestId("account-header").within(() => {
-      cy.findByText(fullName);
-      cy.findByText(email);
-    });
-    cy.findByDisplayValue(first_name);
-    cy.findByDisplayValue(last_name);
-    cy.findByDisplayValue(email);
-    cy.button("Update").should("be.disabled");
+    const page = AccountSettingsPage.visit();
+
+    page.header.verifyFullname(fullName).verifyEmail(email);
+    page
+      .verifyTabSelected("Profile")
+      .tabs.profileForm.verifyValues({
+        firstName: first_name,
+        lastName: last_name,
+        email,
+      })
+      .verifySubmitDisabled();
   });
 
   it("should update the user without fetching memberships", () => {
     cy.intercept("GET", "/api/permissions/membership").as("membership");
-    cy.visit("/account/profile");
-    cy.findByDisplayValue(first_name).click().clear().type("John");
-    cy.findByText("Update").click();
-    cy.findByDisplayValue("John");
+    const page = AccountSettingsPage.visit();
+    page.tabs.profileForm
+      .fill({ firstName: "John" })
+      .submit()
+      .verifyValues({ firstName: "John" });
 
     // It is hard and unreliable to assert that something didn't happen in Cypress
     // This solution was the only one that worked out of all others proposed in this SO topic: https://stackoverflow.com/a/59302542/8815185
     cy.get("@membership.all").should("have.length", 0);
   });
 
+  it("should redirect to the login page when the user has changed the password and logged out (metabase#18151)", () => {
+    const page = AccountSettingsPage.visit();
+    page
+      .selectTab("Password")
+      .tabs.passwordForm.fill({
```

Comment:
Probably it was better before the `tabs`. I don't have any ideas how to make it better atm, maybe we should revert to the previous version without tabs